executor/restore_test.go

```diff
@@ -2,194 +2,181 @@ package executor
 
 import (
 	"context"
-	"strings"
-
-	k8upv1alpha1 "github.com/vshn/k8up/api/v1alpha1"
-	"github.com/vshn/k8up/job"
-
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/ginkgo/extensions/table"
-	. "github.com/onsi/gomega"
-	. "github.com/onsi/gomega/gstruct"
-	gtypes "github.com/onsi/gomega/types"
+	"testing"
 
+	"github.com/stretchr/testify/assert"
 	batchv1 "k8s.io/api/batch/v1"
-	v1 "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/rest"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/envtest"
-)
 
-// These tests use Ginkgo (BDD-style Go testing framework). Refer to
-// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+	k8upv1a1 "github.com/vshn/k8up/api/v1alpha1"
+	"github.com/vshn/k8up/job"
+)
 
 var (
-	restCfg   *rest.Config
-	k8sClient client.Client
-	testEnv   *envtest.Environment
-	scheme    = runtime.NewScheme()
+	testScheme = runtime.NewScheme()
 )
 
-var _ = BeforeSuite(func(done Done) {
-	err := k8upv1alpha1.AddToScheme(scheme)
-	Expect(err).NotTo(HaveOccurred())
+type PVCExpectation struct {
+	Expected  bool
+	ClaimName string
+	ReadOnly  bool
+}
+type ContainerExpectation struct {
+	Expected         bool
+	Image            string
+	Args             []string
+	EnvVars          map[string]string
+	EnvSecretKeyRefs map[string]string
+}
+type VolumeMountExpectation struct {
+	Expected bool
+	Name     string
+	Path     string
+}
 
-	err = batchv1.AddToScheme(scheme)
-	Expect(err).NotTo(HaveOccurred())
+var yes = true
 
-	close(done)
-})
+func Test_buildRestoreObject(t *testing.T) {
+	err := batchv1.AddToScheme(testScheme)
+	if err != nil {
+		t.Errorf("Failed to register batchv1 scheme: %s", err)
+	}
 
-var _ = Describe("Restore", func() {
-	type buildRestoreObjectCase struct {
-		Restore     *k8upv1alpha1.Restore
-		Config      *job.Config
-		ExpectedJob *batchv1.Job
-		JobMatcher  gtypes.GomegaMatcher
+	err = k8upv1a1.AddToScheme(testScheme)
+	if err != nil {
+		t.Errorf("Failed to register k8upv1alpha1 scheme: %s", err)
 	}
-	DescribeTable("buildRestoreObject", func(c buildRestoreObjectCase) {
-		e := NewRestoreExecutor(*c.Config)
-		j, err := e.buildRestoreObject(c.Restore)
 
-		Expect(err).To(BeNil())
-		Expect(j).NotTo(BeNil())
-		Expect(j).To(PointTo(c.JobMatcher))
-	},
-		Entry("builds a job object for S3 restore", buildRestoreObjectCase{
-			Restore: &k8upv1alpha1.Restore{
-				Spec: k8upv1alpha1.RestoreSpec{
-					RestoreMethod: &k8upv1alpha1.RestoreMethod{
-						S3: &k8upv1alpha1.S3Spec{
+	tests := map[string]struct {
```

Comment:
I prefer to have the test tables defined on the package level for two reasons:

1. It makes the test function easier to read. The test table can be a uge distraction herer.
2. The test data can be used in more than one test function. I actually used this quite a lot (for example in two way data conversions but there are also other cases this comes in handy).